python/tskit/trees.py

```diff
@@ -2917,6 +2918,63 @@ def samples(self, population=None, population_id=None):
             samples = samples[sample_population == population]
         return samples
 
+    def write_fasta(self, output, sequence_ids=None, wrap_width=60):
+        """
+        Writes a fasta formatted file from tree sequence data to the
+        specified file-like object.
+
+        Default `sequence_ids` (i.e. the text immediately following ">") are
+        "tsk_{sample_number}" e.g. "tsk_0", "tsk_1" etc. They can be set by providing
+        a list of strings to the `sequence_ids` argument, which must equal the length
+        of the number of samples. Please ensure that these are unique and compatible with
+        fasta standards, since we do not check this (however, do not include the ">").
+        Default `wrap_width` for sequences is 60 characters in accordance with fasta
+        standard outputs, but this can be specified. In order to avoid any line-wrapping
+        of sequences, set `wrap_width = 0`.
+
+        Example usage:
+
+        .. code-block:: python
+            with open("output.fasta", "w") as fasta_file:
+                ts.write_fasta(fasta_file)
+
+        :param File output: The file-like object to write the fasta output.
+        :param list(str) sequence_ids: A list of string names to uniquely identify
+            each of the sequences in the fasta file. Ids will be printed after ">"
+            so you do not need to include this in the strings.
+            If specified, this must be a list of strings of length equal to the number
+            of samples which are output. Note that we do not check the form of these
+            strings in any way, so that it is possible to output bad fasta IDs (for
+            example, by including spaces before the unique identifying part of the
+            string). The default is to output ``tsk_j`` for the jth individual.
+        :param int wrap_width: This parameter specifies the number of sequence
+            characters to include on each line in the fasta file, before wrapping
+            to the next line for each sequence. Defaults to 60 characters in
+            accordance with fasta standard outputs. To avoid any line-wrapping of
+            sequences, set `wrap_width = 0`. Otherwise, supply any positive integer.
+        """
+        # if not specified, IDs default to sample index
+        if sequence_ids is None:
+            sequence_ids = ["tsk_{}".format(j) for j in self.samples()]
+        if len(sequence_ids) != self.num_samples:
+            raise ValueError(
+                "sequence_ids must have length equal to the number of samples.")
+        if type(wrap_width) != int:
```

Comment:
Good idea to check that these are integers, but this is a bit tricky (suppose the input was a numpy integer, e.g.). I'm tempted to let this fall through to ``text_wrap`` --- what happens if we provide a string or double to it?

Reply:
I guess if we want to do it here, the canonical way would be to just convert to int:

```
wrap_width = int(wrap_width)
```
If it's not convertible to int, then it'll throw a type error as we want.

Reply:
Okay, fair enough, I just didn't think that the error output was very nice, it'll look like this if I get rid of the integer check.
![image](https://user-images.githubusercontent.com/37187502/64021747-9167a880-cb2c-11e9-8256-e85b7849ba00.png)

Reply:
Ah this comment didn't load before, which approach do you think is better?

> 
> 
> I guess if we want to do it here, the canonical way would be to just convert to int:
> 
> ```
> wrap_width = int(wrap_width)
> ```
> 
> If it's not convertible to int, then it'll throw a type error as we want.

Reply:
For now I've just deleted the `raise TypeError()` part of the `write_fasta` function (but it's still in the tests). I can add something back in though

Reply:
now added `wrap_width = int(wrap_width)`